src/biome/text/dataset.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import multiprocessing
+import pickle
+from typing import Union, Dict, Iterable, List
+
+import datasets
+from allennlp.data import AllennlpDataset, AllennlpLazyDataset, Instance
+
+InstancesDataset = Union[AllennlpDataset, AllennlpLazyDataset]
+
+
+class Dataset:
```

Comment:
In terms of backward compatibility for data readers. Is there dataset readers for `parquet` or  `excel` formats? Could we create our own data readers easily ?

Reply:
No there is no builtin reader for parquet or excel. It is possible to implement a custom reader but i am not sure if this is straight forward, i haven't had a detailed look at it yet: https://huggingface.co/docs/datasets/master/add_dataset.html

Since we used parquet mainly internally, i think we could switch to their format (arrow format) via `Dataset.save_to_disk`/`Dataset.load_from_disk`. If we want to support excel explicitly, i think we have to implement it ourselves.

Reply:
mkay !